src/clr-angular/data/datagrid/datagrid-row.ts

```diff
@@ -74,6 +75,7 @@ export class ClrDatagridRow<T = any> implements AfterContentInit, AfterViewInit
     public globalExpandable: ExpandableRowsCount,
     public expand: DatagridIfExpandService,
     private displayMode: DisplayModeService,
+    public detailService: DetailService,
```

Comment:
Detail service is also injected into `ClrDatagrid` constructor, doesn't this give us multiple instances of DetailService? e.g 

1. One for ClrDatagrid
2. And one for _every_ ClrDatagridRow